Posts/2021/07/tfl-regkey.md

```diff
@@ -0,0 +1,162 @@
+For the filestore provider, PowerShell provides you with provider drives pointing to the Windows volumes in your system, such as **C:**, **D:**, etc.
+You can also create a provider drive called **DB:** that points to **D:\Dropbox** by using the **New-PSDrive** cmdlet.
+You can persist the drive name by adding the statement to your profile should this be useful.
+
+With the registry provider, PowerShell provides you with two built-in drives: **HKLM:** and **HKCU**.
+The **HKLM:** drive exposes the local machine registry hive - which you (and Windows) use for system wide settings.
+You use the **HKCU:** drive to access the current user's registry hive.
+
+You can discover the provider based drives by using the ``Get-PSProvider`` cmdlet, like this:
+
+```powershell-console
+PS> Get-PSDrive
+
+Name     Used (GB)   Free (GB) Provider      Root
+----     ---------   --------- --------      ----
+Alias                          Alias
+C           262.51      714.58 FileSystem    C:\
+Cert                           Certificate   \
+D          1312.83      596.76 FileSystem    D:\
+db         1312.83      596.76 FileSystem    D:\DropBox
+docs       1312.83      596.76 FileSystem    D:\Dropbox\PACKT…
+Env                            Environment
+F                              FileSystem    F:\
+Function                       Function
+G             2.68       56.79 FileSystem    G:\
+gd         3169.18      556.84 FileSystem    M:\gd
+H          2860.16      865.85 FileSystem    H:\
+HKCU                           Registry      HKEY_CURRENT_USER
+HKLM                           Registry      HKEY_LOCAL_MACHINE..
+```
+Some Windows features come with additional providers, such as the the **ActiveDirectory** RSAT module.
+This feature includes an AD provider:
+
+```powershell-console
+PS> Import-Module -Name ActiveDirectory
+PS> Get-PSProvider -Name ActiveDirectory
+
+Name             Capabilities                                          Drives
+----             ------------                                          ------
+ActiveDirectory  Include, Exclude, Filter, ShouldProcess, Credentials  {AD}
+```
+
+## Registry Value Entries
+
+As I mentioned above, a registry key can contain value entries.
+You can think of each value entry as an attribute of a registry key.
+You use the ``*-ItemProp0erty`` cmdlets to manage individual registry values. 
+But how does this relate to the question?
+Let's begin by looking at the script in question:
+
+```powershell-console
+$RegistryPath = 'HKCU:\Software\CommunityBlog\Scripts'
+$Name         = 'Version'
+$Value        = '42'
+New-ItemProperty -Path $RegistryPath -Name $Name -Value $Value -PropertyType DWORD -Force 
+
+New-ItemProperty: Cannot find path 'HKCU:\Software\CommunityBlog\Scripts' because it does not exist.
+```
+
+The script used the `New-ItemProperty` to create a Version value entry to a specific key.
+This script, however, fails since the registry key, specified in **$RegistryPath** variable does not exist.
```

Comment:
```suggestion
The script used the `New-ItemProperty` to create a **Version** value entry to a specific key.
This script, however, fails since the registry key, specified in `$RegistryPath` variable does not exist.
```